### PR TITLE
New addon: Customizable profile picture outline

### LINKED ISSUE #2697: Remove border on pfps
### Is your request related to a problem? Please describe.

Scratch recently added this cool epic amazing looking (/s) border on profile pictures on project comments.
Screen Shot 2021-06-06 at 8 47 41 PM

### Describe the solution you'd like

override it in css

### Describe alternatives you've considered

idk maybe add a different type of border

### Additional context

there we go, I got the link. there's a [forum topic](https://scratch.mit.edu/discuss/topic/517693/) about the border on pfps, and it's pretty unpopular

(I see you fdreerf! You think you can downvote my posts..)

### PR DESCRIPTION
**Pull request by CST1229**
Resolves #2697

### Changes

Adds the `customize-avatar-border` addon, which allows you to hide the outline around profile pictures in 3.0-style comments, alongside changing the color or making it fill behind the picture..

### Reason for changes

The outline looks kinda bad on transparent profile pictures so it's nice to hide it or make it extend behind it.

### Tests

Seems to work.
hide
fill
color

### CODE REVIEW

File: addons/customize-avatar-border/addon.json [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      "link": "https://github.com/CST1229"
+    }
+  ],
+  "versionAdded": "1.18.0",
+  "dynamicEnable": true,
+  "dynamicDisable": true,
+  "updateUserstylesOnSettingsChange": true,
+  "injectAsStyleElt": true,
+  "enabledByDefault": false
+}
```
Comment (FunctionalMetatable) [inline addons/customize-avatar-border/addon.json:64]: please clean up the manifest for the sake of readers
```suggestion
{
  "name": "Customizable profile picture outline",
  "description": "Allows you to hide or change the outline around profile pictures in comments in 3.0-styled pages.",
  "credits": [
    {
      "name": "CST1229",
      "link": "https://github.com/CST1229"
    }
  ],
  "userstyles": [
    {
      "url": "hide_outline.css",
      "matches": ["https://scratch.mit.edu/projects/*", "https://scratch.mit.edu/studios/*"],
      "settingMatch": {
        "id": "hide-outline",
        "value": true
      }
    },
    {
      "url": "outline_color.css",
      "matches": ["https://scratch.mit.edu/projects/*", "https://scratch.mit.edu/studios/*"],
      "settingMatch": {
        "id": "hide-outline",
        "value": false
      }
    },
    {
      "url": "fill_transparent.css",
      "matches": ["https://scratch.mit.edu/projects/*", "https://scratch.mit.edu/studios/*"],
      "settingMatch": {
        "id": "fill-transparent",
        "value": true
      }
    }
  ],
  "dynamicEnable": true,
  "dynamicDisable": true,
  "updateUserstylesOnSettingsChange": true,
  "injectAsStyleElt": true,
  "settings": [
    {
      "name": "Hide outline",
      "id": "hide-outline",
      "type": "boolean",
      "default": false
    },
    {
      "name": "Fill transparency",
      "id": "fill-transparent",
      "type": "boolean",
      "default": false
    },
    {
      "name": "Outline colour",
      "id": "outline-color",
      "type": "color",
      "allowTransparency": true,
      "default": "#4d97ff40"
    }
  ],
  
  "tags": ["community", "theme"],
  "versionAdded": "1.18.0",
  "enabledByDefault": false
}
```

**Review by WorldLanguages — APPROVED** [review]

**Comment by WorldLanguages** [comment]
@mxmou Can you approve this? This new addon will likely need to be kept in mind within CSS priority, in case other themes modify the box-shadow of background-color of comment profile pictures.

File: addons/customize-avatar-border/addon.json [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  "injectAsStyleElt": true,
+  "settings": [
+    {
+      "name": "Hide outline",
+      "id": "hide-outline",
+      "type": "boolean",
+      "default": false
+    },
+    {
+      "name": "Fill transparency",
```
Comment (mxmou) [inline addons/customize-avatar-border/addon.json:48]: This name isn't very descriptive.
```suggestion
      "name": "Fill transparent profile pictures",
```

**Review by mxmou — COMMENTED** [review]
Some suggestions

File: addons/addons.json [hunk 3]
```diff
@@ -97,6 +97,7 @@
   "editor-theme3",
   "// For website:",
   "scratchr2",
+  "customize-avatar-border",
```
[the remark this answers is not in the document]
Comment (CST1229) [reply addons/addons.json]: I didn't really think of that and I just put it somewhere.

**Review by mxmou — APPROVED** [review]

**Pull request merged**